src/NuGet.Core/NuGet.Build.Tasks/RestoreTaskEx.cs

```diff
@@ -243,7 +247,11 @@ private IEnumerable<string> GetCommandLineArguments()
             }
 #endif
             msBuildGlobalProperties["ExcludeRestorePackageImports"] = "true";
-            msBuildGlobalProperties["SolutionPath"] = SolutionPath;
+
+            if (IsSolutionPathDefined)
+            {
+                msBuildGlobalProperties["SolutionPath"] = SolutionPath;
+            }
 
             return msBuildGlobalProperties;
```

Comment:
Would it be possible to add unit tests?

Reply:
I have another change in the works to add a new unit test assembly for my new console app.  I'll add the unit test in that PR so I don't have to go through CI again for this one.